firexapp/firex_subprocess.py

```diff
@@ -492,6 +511,13 @@ def _collect_stats():
 
                 # Inactivity timeout check
                 current_output_size = os.fstat(f.fileno()).st_size
+                for file in found_monitor_activity_files:
+                    try:
+                        with open(file, 'r') as fp:
+                            current_output_size += os.fstat(fp.fileno()).st_size
+                    except Exception as e:
```

Comment:
Ideally we catch the most narrow relevant exceptions, I think `OSError` should be sufficient if `found_monitor_activity_files` is always a list of strings.

Reply:
Updated 👍

Reply:
I know I keep repeating this, but it's worth repeating :-)
Umer, note that this isn't only "ideally", but any "except Exception" in our code is broken, and it's particularly bad here. This is inside the "wait for other processes" loop, which is where most of our SoftTimeLimit timeouts occur. This clause would swallows those timeouts, which means it was quite likely to break our SoftTimeLimit timeouts in some cases. This is Bad (TM).

Moral of the story: "except Exception" does not belong in (most!) Python code at all, and is particularly prone to breaking celery code.

Reply:
While I'm looking at this: If we somehow cannot stat that file, we do not want to print an error every half second or whatever it is, which I think this does. Whoever needs to look at those logs will not have kind thoughts about whoever put this in :-)

You might want a sentinel "already printed the error" or some such method.

Reply:
> I know I keep repeating this, but it's worth repeating :-) Umer, note that this isn't only "ideally", but any "except Exception" in our code is broken, and it's particularly bad here. This is inside the "wait for other processes" loop, which is where most of our SoftTimeLimit timeouts occur. This clause would swallows those timeouts, which means it was quite likely to break our SoftTimeLimit timeouts in some cases. This is Bad (TM).
> 
> Moral of the story: "except Exception" does not belong in (most!) Python code at all, and is particularly prone to breaking celery code.

Understood and noted in bold!

Reply:
> While I'm looking at this: If we somehow cannot stat that file, we do not want to print an error every half second or whatever it is, which I think this does. Whoever needs to look at those logs will not have kind thoughts about whoever put this in :-)
> 
> You might want a sentinel "already printed the error" or some such method.

Good point updated accordingly to not repeat logging